contrib/sarplus/python/pysarplus/SARPlus.py

```diff
@@ -171,42 +180,54 @@ def fit(self, df):
             GROUP BY i1, i2
             HAVING value >= {threshold}
             CLUSTER BY i1, i2
-        """)
+        """
+        )
 
         item_cooccurrence = self.spark.sql(query)
         item_cooccurrence.write.mode("overwrite").saveAsTable(
             self._format("{prefix}item_cooccurrence")
         )
 
+        # compute item frequencies
+        self.item_frequencies = item_cooccurrence.filter(
+            F.col("i1") == F.col("i2")
+        ).select(F.col("i1").alias("item_id"), F.col("value").alias("frequency"))
+
```

Comment:
question to @simonzhaoms will the item-item function have any relationship with the work you are doing about the new similarities?